Competition2017/src/competition/subsystems/drive/commands/DriveForDistanceAtHeadingCommand.java

```diff
@@ -0,0 +1,141 @@
+package competition.subsystems.drive.commands;
+
+import java.util.function.DoubleSupplier;
+
+import com.google.inject.Inject;
+import xbot.common.logging.RobotAssertionManager;
+import xbot.common.math.ContiguousHeading;
+import xbot.common.math.PIDFactory;
+import xbot.common.math.PIDManager;
+import xbot.common.math.XYPair;
+import xbot.common.properties.DoubleProperty;
+import xbot.common.properties.XPropertyManager;
+import competition.subsystems.drive.DriveSubsystem;
+import competition.subsystems.pose.PoseSubsystem;
+
+public class DriveForDistanceAtHeadingCommand extends BaseDriveCommand {
+    protected final DoubleProperty speedHeadingThresh;
+    protected final PoseSubsystem poseSubsystem;
+    protected ContiguousHeading targetHeading;
+    protected double targetDistance;
+
+    protected DoubleSupplier targetHeadingSupplier;
+    protected DoubleSupplier targetDistanceSupplier;
+    
+    private final PIDManager headingDrivePid;
+    private final PIDManager distancePid;
+    private final PIDManager alignmentPid;
+    
+    private XYPair initialPosition = null;
+    
+    @Inject
+    public DriveForDistanceAtHeadingCommand(
+            DriveSubsystem driveSubsystem, 
+            XPropertyManager propMan,
+            PoseSubsystem pose,
+            RobotAssertionManager assertionManager,
+            PIDFactory pidFactory)
+    {
+        super(driveSubsystem);
+        this.poseSubsystem = pose;
+        speedHeadingThresh = propMan.createPersistentProperty("Drive at heading threshold for translation", 5);
+        
+        headingDrivePid = new PIDManager("Drive at heading rotation", propMan, assertionManager,  1 / 80d, 0, 0);
```

Comment:
I just inherited this. I would _rather_ not mess with it because the order of the parameters are different with the PID manager, and we don't have any way of proving that the new options work given that those properties already exist.